ministryofjustice/cloud-platform-cli
# Create functionality to amend the users Elasticseach module

Overview
---

This PR connects to https://github.com/ministryofjustice/cloud-platform/issues/3158 and relates to the requirement to amend users with an Elasticsearch module in Terraform HCL. To migrate users need to add two attributes.

When a user migrates, the cli will check for the existance of an Elasticseach module. If found, the `changeElasticSearch` function will be called and will append a new version of the elasticsearch block to the end of said elasticsearch file.

How do I test this PR
---

- Build the cli: clone the repo and run `make` at the root of the repo.
- Using the newly built CLI, clone or cd into the cloud-platform-environments repository. Find a namespace with an elasticsearch module and run the command: `cloud-platform environment migrate`.
- If you run `git status` it should show a change to the `resources/elasticsearch.tf` file.

Or

- Just run `go test -v ./...` 😄
